titiler/pgstac/factory.py

```diff
@@ -555,6 +557,18 @@ def wmts(
             if qs:
                 tiles_url += f"?{urlencode(qs)}"
 
+            dep = get_dependant(path="", call=self.layer_dependency)
+            query_values, _ = request_params_to_args(dep.query_params, QueryParams(qs))
```

Comment:
1. we first get the `dependencies` for the specific function (here it should be a class https://github.com/developmentseed/titiler/blob/main/src/titiler/core/titiler/core/dependencies.py#L140) 

2. we use `request_params_to_args` to translate the query parameters to a dictionary which we will use as argument to initiate the `layer_dependency` class

3. we catch the exception raised by the `layer_dependency` and if there is no `layers` we raise it back 

4. if no errors we add a `default` layer to the list of layers